src/ui/component/copyToClipboard/copyToClipboard.tsx

```diff
@@ -1,32 +1,59 @@
+import { useCallback, useState } from 'react'
 import { useAppStore } from '@/ui/store/appStore'
-import { useCallback } from 'react'
+import { useDebounce } from '@/ui/hook/useDebounce'
 import { Icon } from '@/ui/component/icon/icon'
+import classNames from 'classnames'
 import './copyToClipboard.scss'
 
+type CopyState = 'copy' | 'success' | 'error'
+
 type CopyToClipBoardProps = {
   textToCopy: string
+  animationDelay?: number
   onCopied?: (isCopied: boolean) => void
 }
 
-export const CopyToClipboard = ({ onCopied, textToCopy }: CopyToClipBoardProps): JSX.Element => {
+export const CopyToClipboard = ({
+  animationDelay = 2000,
+  onCopied,
+  textToCopy
+}: CopyToClipBoardProps): JSX.Element => {
   const theme = useAppStore(state => state.theme)
+  const [copyState, setCopyState] = useState<CopyState>('copy')
+
+  const resetCopyState = useCallback((): void => {
+    setCopyState('copy')
+  }, [])
```

Comment:
This is not enough for me 😌

My comments on previous PR about having a hook that handle the copy logic is not implemented here.

All the `copyToClipboard` logic should be abstracted in a hook that will handle the state and the handler function. The component should not have to worry about that, to be dumb and generic as much as possible ;)

Reply:
`<CopyToClipboard />` now checks the `copyState` from the `useCopyToClipboard` hook to know which icon to show, and onClick I use the `handleCopy` that returns a promise telling if the copy succeeded or not.